presto-hive/src/main/java/com/facebook/presto/hive/HiveType.java

```diff
@@ -283,4 +286,10 @@ public static Type getPrimitiveType(PrimitiveTypeInfo typeInfo)
                 return null;
         }
     }
+
+    public int getEstimatedRetainedSizeInBytes()
```

Comment:
after in person conversation it's because the values are shared from a cache.  So the plan is
1. add a note explaining.
2. get rid of dead code in HiveTypeName if HiveTypeName.getEstimatedSizeInBytes isn't used anywhere else

Reply:
The class has two filelds

```
private final HiveTypeName hiveTypeName;
private final TypeInfo typeInfo;
```

`hiveTypeName` is set from `typeInfo`

`TypeInfo` objects are always created by the `TypeInfoFactory` that has a static singleton cache. Thus it doesn't make much sense to account memory for these objects here, as those are shared.

Reply:
> hiveTypeName is set from typeInfo

Sorry, that is not true. HiveTypeName is created based on `TypeInfo`. We still need to account for it. Let me fix it.

Reply:
> Wow, does that mean Column is not counted for memory usage for years?

In this PR the `Map<Integer, HiveTypeName> columnCoercion` is replaced with the `Map<Integer, Column>`, that's why the size of the `Column` has to be accounted.

> BTW: why the method name is not getEstimatedRetainedSizeInBytes?

I believe the question is why it is not the `getEstimatedSizeInBytes`. I wanted the name to be kinda descriptive of the fact that the size of the `TypeInfo` is not accounted, as it is not "retained" by the Column object.